crates/ty_python_semantic/src/types/function.rs

```diff
@@ -616,10 +616,10 @@ impl<'db> FunctionLiteral<'db> {
         // We only include an implementation (i.e. a definition not decorated with `@overload`) if
         // it's the only definition.
         let (overloads, implementation) = self.overloads_and_implementation(db);
-        if let Some(implementation) = implementation {
-            if overloads.is_empty() {
-                return CallableSignature::single(implementation.signature(db));
-            }
+        if let Some(implementation) = implementation
+            && overloads.is_empty()
```

Comment:
It will take me a while to get used to this syntax. My brain immediately raises a syntax error after `implementation` that there's a missing `{` 😅